### PR TITLE
NCS-743 Use api security java library

### PR DESCRIPTION
**Pull request by markpit**
Changing the endpoint authorisation interceptors to use the interceptors from api-security-java.

When a user logs in, they have permission in their ERIC header that are related to their scope. If they authenticate against a company using the company auth code, then they also get company permissions added into their ERIC headers when they make requests to the endpoints. 
We are using the CRUD authentication interceptor(for user auth checks and company auth checks) and Internal user interceptor (for api key access).
The CRUD interceptor will match Create, Read , Update and delete access against GET, POST, PUT, DELETE api calls.
The permission given to the user are set in here ->
https://github.com/companieshouse/account.ch.gov.uk/blob/master/lib/AccountChGovUk/Helpers/ScopeMapper.pm#L166
We're adding a new company_confirmation_statement = create,read,update  permission (no delete) for company permission and using the default user_profile = read permission that the user will have by default.

// These are called before the user authenticates against the company. 
// These use CRUDInterceptor with USER_PROFILE permission key
GET /confirmation-statement/company/{companyNumber}/next-made-up-to-date
GET /confirmation-statement/company/{company-number}/eligibility

// These are using InternalUserInterceptor as they are called via apis
GET /transactions/{transaction_id}/confirmation-statement/{confirmation-statement-id}/costs
GET /private/transactions/{transaction_id}/confirmation-statement/{confirmation_statement_id}/filings

// These are using CRUD interceptor with new COMPANY_CONFIRMATION_STATEMENT permission key (i.e. if user is authenticated against the company)
POST /transactions/{transaction_id}/confirmation-statement/
POST /transactions/{transaction_id}/confirmation-statement/{confirmation_statement_id}
GET /transactions/{transaction_id}/confirmation-statement/{confirmation_statement_id}
GET /transactions/{transaction_id}/confirmation-statement/{confirmation_statement_id}/validation-status
GET /confirmation-statement/company/{companyNumber}/active-director-details
GET /confirmation-statement/company/{companyNumber}/persons-of-significant-control
GET /confirmation-statement/company/{companyNumber}/register-locations
GET /confirmation-statement/company/{companyNumber}/shareholders
GET /confirmation-statement/company/{companyNumber}/statement-of-capital

Changes to account.ch.gov.uk to add the company permissions for confirmation statement - https://github.com/companieshouse/account.ch.gov.uk/pull/392

JIRA Story - https://companieshouse.atlassian.net/browse/NCS-743

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -88,6 +88,11 @@
             <version>${sonar-maven-plugin.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>uk.gov.companieshouse</groupId>
+            <artifactId>structured-logging</artifactId>
```
Comment (ICoxCH) [inline pom.xml:93]: why do we need structured logging?

Comment (markpit) [reply pom.xml:93]: It's used by the InternalUserInterceptor that we import from api-security-java (and maybe CRUD one as well?) but the api-security-java pom has it as an optional dependency meaning that the importing pom has to supply the dependency
https://github.com/companieshouse/api-security-java/blob/master/pom.xml#L64

Comment (ICoxCH) [reply pom.xml:93]: this seems to me as not how use the optional tag, 
Does importing this here interfere with our current log structure at all?

**Review by ICoxCH — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (markpit) [reply pom.xml:93]: we've noticed that importing structured logging means we lose the application name and class name from the logs. I'll raise a separate jira story to integrate with structured logging.

Comment (markpit) [reply pom.xml:93]: Jira story to use structured-logging - https://companieshouse.atlassian.net/browse/NCS-756

**Pull request merged**